### PR TITLE
Add Op (aten::flatten) | feat(torchlib)

### PR DESCRIPTION
**Pull request by titaiwangms**
Add aten::flatten based on torchscript implementation

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
Thanks! Does this help with some models?

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -3492,6 +3492,49 @@ def aten_fix(self: TensorType) -> TensorType:
...
+    elif start_dim == 0:
+        if end_dim in (-2, dim - 2):
+            return op.Flatten(self, axis=end_dim + 1)
+
+    # if end_dim is negative add dim
+    if end_dim < 0:
+        end_dim = dim + end_dim
+
+    input_size = op.Shape(self)
+    slice1 = op.Slice(
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:3514]: What is this slice? Consider a better name?

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 2]
```diff
@@ -3492,6 +3492,49 @@ def aten_fix(self: TensorType) -> TensorType:
...
+        input_size,
+        op.Constant(value_ints=[0]),
+        op.Constant(value_ints=[start_dim]),
+        op.Constant(value_ints=[0]),
+    )
+    slices = [slice1, op.Constant(value_ints=[-1])]
+    if end_dim < dim - 1:
+        slice3 = op.Slice(
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:3522]: Is there a slice2?

**Comment by justinchuby** [comment]
@titaiwangms some follow up comments

**Comment by titaiwangms** [comment]
> Thanks! Does this help with some models?

Just saw in torchscript, aten::repeat_interleave using aten::flatten, and it looks easy to implement.

**Comment by titaiwangms** [comment]
> @titaiwangms some follow up comments

https://github.com/microsoft/onnxscript/pull/1809

**Pull request merged**